pyccel/ast/core.py

```diff
@@ -2065,6 +2066,41 @@ def _ignore(cls, c):
         """
         return c is None or isinstance(c, (FunctionDef, *cls._ignored_types))
 
+class IndexedFunctionCall(FunctionCall):
+    """
+    Represents an indexed function call in the code.
+
+    Class representing indexed function calls, encapsulating all
+    relevant information for such calls within the code base.
+
+    Parameters
+    ----------
+    func : FunctionDef
+        The function being called.
+
+    args : list of FunctionCallArgument
+        The arguments passed to the function.
+
+    indexes : list of TypedAstNode
+        The indexes of the function call.
+
+    current_function : FunctionDef, optional
+        The function where the call takes place.
+    """
+    __slots__ = ('_indexes',)
+    def __init__(self, func, args, indexes, current_function = None):
+        self._indexes = indexes
```

Comment:
These indices are Pyccel objects so they should be saved in `_attribute_nodes`

```suggestion
    __slots__ = ('_indexes',)
    _attribute_nodes = FunctionCall._attribute_nodes + ('_indexes',)
    def __init__(self, func, args, indexes, current_function = None):
        self._indexes = indexes
```

Reply:
[84c1b53](https://github.com/pyccel/pyccel-cuda/pull/42/commits/84c1b535cf9ce3a5c5cfc204c2d9da5abdd98843)